rustls/src/client/hs.rs

```diff
@@ -359,11 +387,24 @@ fn emit_client_hello_for_retry(
         payload: HandshakePayload::ClientHello(chp_payload),
     };
 
-    let early_key_schedule = if let Some(resuming) = tls13_session {
-        let schedule = tls13::fill_in_psk_binder(&resuming, &transcript_buffer, &mut chp);
-        Some((resuming.suite(), schedule))
-    } else {
-        None
+    let early_key_schedule = match (ech_state.as_mut(), tls13_session) {
+        // If we're performing ECH and resuming, then the PSK binder will have been dealt with
+        // separately and we need to take the early_data_key_schedule computed for the inner hello.
+        // TODO(@cpu): This is an area I'm not 100% confident the implementation is correct!
```

Comment:
is this something that has been addressed now, or is it still a worry? any ideas on how to gain some confidence?

Reply:
Oops, this slipped through from an earlier revision. I'm confident in the implementation now (and bogo concurs). I will remove the comment.

Reply:
Removed stale TODO.